common/eventconsumption/eventvisitor.h

```diff
@@ -17,7 +17,49 @@
 
 #pragma once
 
+#include "eventconsumption.h"
 // MORE: If jevent.hpp replaces readEvents with a function that returns a reader instance, from
 // which events are pulled instead of visisted, then this file will declare the IEventVisisot
 // interface and the adapter to convert pulled events into visisted events.
 #include "jevent.hpp"
+
+// Abstraction of the visitor pattern for "reading" binary event data files. The `readEvents`
+// function will pass each byte of data contained within a file throwgh exactly one method of
+// this interface.
+//
+// For each compatibile file the visitor can expect:
+// 1. One call to visitFile
+// 2. Zero or more sequences of:
+//    a. One call to visitEvent
+//    b. Zero or more calls to visitAttribute
+//    c. One call to departEvent
+// 3. One call to departFile
+//
+// Implementations may implement limited filtering during visitation. All methods, except
+// `departFile`, may abort visitation. Both `visitEvent` and `visitAttribute` (in the context of
+// an event) may suppress visitation of the remainder of the current event.
+//
+// Reasons for aborting a file include:
+// - unrecognized file version; or
+// - (remaining) events out of a target date range; or
+// - trace IDs are required but are not present.
+//
+// Reasons for suppressing an event include:
+// - event type not required by current use case; or
+// - attribute value not out of range for current use case.
+interface IEventAttributeVisitor : extends IEventVisitor
```

Comment:
I don't think this is the correct structure.  Better is to have separate interfaces - i.e. not derive this from IEventVisitor.  Otherwise it is not clear which functions need to be implemented by an interface.

Then have a class that implements an IEventVisitor and plays the output to an IEventAttributeVisitor.